src/ceph-disk/ceph_disk/main.py

```diff
@@ -0,0 +1,5828 @@
+            else:
+                LOG.warning('%s %s was not prepared with '
+                            'ceph-disk. Symlinking directly.',
+                            self.name.capitalize(),
+                            getattr(self.args, self.name))
+                self.space_symlink = getattr(self.args, self.name)
+                return
+
+        self.space_symlink = '/dev/disk/by-partuuid/{uuid}'.format(
+            uuid=getattr(self.args, self.name + '_uuid'))
+
+        if self.args.dmcrypt:
+            self.space_dmcrypt = self.space_symlink
+            self.space_symlink = '/dev/mapper/{uuid}'.format(
+                uuid=getattr(self.args, self.name + '_uuid'))
+
+        if reusing_partition:
+            # confirm that the space_symlink exists. It should since
+            # this was an active space
+            # in the past. Continuing otherwise would be futile.
+            assert os.path.exists(self.space_symlink)
+            return
+
+        num = self.desired_partition_number()
+
+        if num == 0:
+            LOG.warning('OSD will not be hot-swappable if %s '
+                        'is not the same device as the osd data',
+                        self.name)
+
+        device = Device.factory(getattr(self.args, self.name), self.args)
+        num = device.create_partition(
+            uuid=getattr(self.args, self.name + '_uuid'),
+            name=self.name,
+            size=self.space_size,
+            num=num)
+
+        partition = device.get_partition(num)
+
+        LOG.debug('%s is GPT partition %s',
+                  self.name.capitalize(),
+                  self.space_symlink)
+
+        if isinstance(partition, DevicePartitionCrypt):
+            partition.format()
+            partition.map()
+
+        command_check_call(
+            [
+                'sgdisk',
+                '--typecode={num}:{uuid}'.format(
+                    num=num,
+                    uuid=partition.ptype_for_name(self.name),
+                ),
+                '--',
+                getattr(self.args, self.name),
+            ],
+        )
+
+        update_partition(getattr(self.args, self.name), 'prepared')
+
+        if self.wipe_device:
+            LOG.debug('Erasing partition %s',
+                      self.space_symlink)
+            command(
+                [
+                    'dd',
+                    'if=/dev/zero',
+                    'of=' + self.space_symlink,
+                ],
+            )
+
+        LOG.debug('%s is GPT partition %s',
+                  self.name.capitalize(),
+                  self.space_symlink)
+
+
+class PrepareJournal(PrepareSpace):
+
+    def __init__(self, args):
+        self.name = 'journal'
+        # ceph-osd now tries to identify the OSD type by checking 
+        # the disk, but at this stage we did not create anything
+        # because of that we assume the values for a Filestore OSD
+        self.allows_journal = True
```

Comment:
To summarize ceph-disk makes three calls to ceph-osd and each of these calls should be delegated to the functions wants_journal, allows_journal and needs_journal of the ObjectStore class.

I had some problems making these calls to work, probably because just ceph-disk used this before its removal.

What I did to solve this was checking the implementations of these functions on FileStore.h
https://github.com/starlingx-staging/stx-ceph/blob/83c70bf033cd6f7b9ea1a1f96023dcfcb912f977/src/os/filestore/FileStore.h#L488-L496

And used those values directly int the class PrepareJournal that is only used by PrepareFilestore on ceph-disk.